games-emulation/genesis_plus_gx_libretro/genesis_plus_gx_libretro-1.7.4_20180303.recipe

```diff
@@ -0,0 +1,52 @@
+SUMMARY="A port of Genesis Plus GX, a Sega 8/16bit emulator to the libretro API"
+DESCRIPTION="Genesis Plus GX is a Sega Genesis, Mega-CD, Game Gear, SG-1000 \
+and Master System (Mark-III) emulator. Its main focuses are accuracy and \
+performance. It is based off the original Genesis Plus by Charles McDonald."
+HOMEPAGE="https://github.com/ekeeke/Genesis-Plus-GX"
+COPYRIGHT="1998-2018 Charles McDonald, Eke-eke, the libretro team"
+LICENSE="GenesisPlusGX"
```

Comment:
Hmm, this is a concatenation of several licenses.
I don't know if this is OK. I think you will need to split the file into several files, one per license.
Then you will need to attach only thoses files which are not already shipped in Haiku.
To see if a license is part of haiku you can use the `catattr` command. For instance:
```
catattr SYS:PACKAGE_FILE /system/data/licenses/GNULIB 
/system/data/licenses/GNULIB : string : gnulib-2013_12_17-1-x86_gcc2.hpkg
```
(Of course, if the answer is `haiku-r1~alpha4_pm_hrevXXXXX-Y-x86_64.hpkg` then this means the license file is shipped in Haiku.)
Sorry for asking for the split...

BTW, the recipe builds and installs fine, and I can see it with retroarch, so once the license file is split we can merge the PR!

Reply:
Sure thing, will do. Thanks for the review.

Reply:
How does it look now @fbrosson ?

Reply:
I'm so sorry, I had not noticed that there were so many licenses!
Well, GenesisPlusGX and Xiph are OK. But I think we shoud:
* rename `LIBCHDR` to `NAME CHD`
* add files for
  - `NTSC Filter Library`
  - `BLIP Buffer Library`
  - `LIBFAT`
  - `LIBASND`
  - `LIBASND`
  - `FIX94`

This is based on the contents of your previous GenesisPlusGX, and I suggest that we put this in the recipe:
```
LICENSE="GenesisPlusGX
	GNU LGPL v2.1
	Xiph
	NAME CHD
	FLAC
	Zlib
	NTSC Filter Library
	BLIP Buffer Library
	LibOGG
	LIBFAT
	LibPNG
	LIBASND
	FIX94"
```
(Hmm, media-libs/flac itself does not even ship the FLAC license, so we might need to also add a `FLAC` file in `licenses`.)

And maybe we could add a `readme-licenses.txt` in `additional-files` with the following content (and install it in `$docDir`):

* TREMOR library is distributed under the following license:
  - Xiph

* LIBCHDR is distributed with code under following licenses:
  - MAME CHD interface
  - FLAC - Free Lossless Audio Codec
  - ZLIB DATA COMPRESSION LIBRARY
  - NTSC FILTER LIBRARY
  - BLIP BUFFER LIBRARY

* Gamecube & Wii ports are linked with LIBOGC library which is distributed under
the following license:
  - LibOGG

* Gamecube & Wii ports are linked with LIBFAT library which is distributed under the following license:
  - LIBFAT

* Gamecube & Wii ports are linked with LIBPNG library which is distributed under the following license:
  - LibPNG

* Gamecube & Wii ports are linked with LIBASND library which is distributed under the following license:
  - LIBASND

* Wii port is linked with WiiDRC library which is distributed under the following license:
  - FIX94


_Well, this is just a suggestion. After all, if everyone agrees with having a monolithic file then I'm OK with reverting to what you had in the first place._

Reply:
I think it's fine to leave LIBCHDR under this name as it's what is actually distributed: the libchdr is used to read MAME CHD file formats. Regarding FLAC, it's using the Xiph-style license so we should be fine (and I, too, looked into media-lib/flac to see what to do).
I added libPNG but it might not be necessary. As the doc says, the Gamecube & Wii ports link with them, but not the Haiku port. Same with LibFAT, libasnd, and FIX94.

Reply:
It's a bit of a mess and there seems to have been disagreements between the Libretro team and the upstream maintainer of Genesis Plus GX even. MAME is now dual licensed under the GPL so it's possible we can remove the LIBCHDR part. BLIP and NTSC are LGPL. FLAC follows the Xiph model as TREMOR does (I ended up creating that Xiph file after not finding any license with Haiku's FLAC packages).

Anything that Gamecube & Wii builds link with should be irrelevant since we're not building on these targets... no?

Reply:
I think we can merge. Thanks for the update and explanation.
(If you are OK with the licenses then I'm OK too :)